### PR TITLE
Add auto name generation

### PR DESCRIPTION
**Pull request by jxstxn1**
Added acronym recommendation

### CODE REVIEW

File: sidekick/lib/src/init/init_command.dart [hunk 1]
```diff
@@ -43,10 +45,37 @@ class InitCommand extends Command {
...
+            lastPart.toAcronym(splitSyllables: true),
+            parentLastPart.toAcronym(),
+            parentLastPart.toAcronym(splitSyllables: true),
+          ];
+          print('Type the number of the fitting acronym');
+          for (int i = 0; i < acronymSuggestions.length; i++) {
+            print('${i + 1}: ${acronymSuggestions[i]}');
+          }
+          print('${acronymSuggestions.length + 1}: Enter your own');
+          final String answer = dcli.ask('> ');
```
Comment (passsy) [inline sidekick/lib/src/init/init_command.dart:67]: Use [`dcli.menu`](https://pub.dev/documentation/dcli/0.21.1/dcli/menu.html) 😉

File: sidekick/lib/src/init/init_command.dart [hunk 2]
```diff
@@ -43,10 +45,37 @@ class InitCommand extends Command {
...
+          final String lastPart = pathParts.last;
+          final String parentLastPart =
+              '${pathParts[pathParts.length - 2]}/$lastPart';
+          print(parentLastPart);
+          final List<String> acronymSuggestions = [
+            lastPart.toAcronym(),
+            lastPart.toAcronym(splitSyllables: true),
+            parentLastPart.toAcronym(),
+            parentLastPart.toAcronym(splitSyllables: true),
+          ];
```
Comment (passsy) [inline sidekick/lib/src/init/init_command.dart:61]: Remove duplicates (`.toSet().toList()`)

**Comment by jxstxn1** [comment]
Im quite unhappy with the performance, will invest some time at the weekend to speed this up

File: sidekick/lib/src/init/name_suggester.dart [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+      suggestions.add(name.toAcronym(splitSyllables: true));
+    }
+
+    // first last character variant
+    final chars = dirName.characters.where((it) => it.isAscii);
+    if (chars.length >= 2) {
+      suggestions.add('${chars.first}${chars.last}');
+    }
+
+    // TODO acronym should return the filtered syllableWords. We could suggest all of them on their own
```
Comment (passsy) [inline sidekick/lib/src/init/name_suggester.dart:73]: Often, project names already consist of acronyms. i.e. we should suggest `dfb` for `dfb-mobile`

**Review by passsy — APPROVED** [review]

File: sidekick/lib/src/init/name_suggester.dart (hunk 3 above)
Comment (jxstxn1) [reply sidekick/lib/src/init/name_suggester.dart:73]: I'll update the acronym package with a short phrase flag

**Pull request merged**